src/Microsoft.ML.Core/Data/ColumnType.cs

```diff
@@ -789,6 +793,19 @@ public override bool Equals(ColumnType other)
             return true;
         }
 
+        public override bool Equals(Object other)
```

Comment:
>Equals [](start = 29, length = 6)

what about all other PrimitiveTypes ? Like TextType, NumberType, etc?

Reply:
I added a comment with an explanation in the ColumnType class.

Reply:
I added a comment with an explanation in the ColumnType class, line 245. basically no need to override them for these classes

---
In reply to: [206708185](https://github.com/dotnet/machinelearning/pull/620#discussion_r206708185) [](ancestors = 206708185)